src/gwt/src/org/rstudio/core/client/hyperlink/RunHyperlink.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * RunHyperlink.java
+ *
+ * Copyright (C) 2022 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+package org.rstudio.core.client.hyperlink;
+
+import com.google.gwt.user.client.ui.HTML;
+import com.google.gwt.user.client.ui.Label;
+import com.google.gwt.user.client.ui.VerticalPanel;
+import com.google.gwt.user.client.ui.Widget;
+
+import org.rstudio.core.client.StringUtil;
+import org.rstudio.core.client.regex.Match;
+import org.rstudio.core.client.regex.Pattern;
+import org.rstudio.studio.client.RStudioGinjector;
+import org.rstudio.studio.client.application.events.EventBus;
+import org.rstudio.studio.client.common.codetools.RCompletionType;
+import org.rstudio.studio.client.server.Server;
+import org.rstudio.studio.client.server.ServerError;
+import org.rstudio.studio.client.server.ServerRequestCallback;
+import org.rstudio.studio.client.workbench.views.console.events.SendToConsoleEvent;
+import org.rstudio.studio.client.workbench.views.help.model.HelpInfo;
+
+public class RunHyperlink extends Hyperlink
+{
+    public RunHyperlink(String url, String params, String text, String clazz)
+    {
+        super(url, params, text, clazz);
+        code_ = url.replaceFirst("^(ide|rstudio):run:", "");
+        Match match = HYPERLINK_PATTERN.match(code_, 0);
+        if (match == null)
+        {
+            package_ = null;
+            allowed_ = false;
+        }
+        else 
+        {
+            package_ = match.getGroup(1);
+            fun_ = match.getGroup(2);
+            allowed_ = true;
+        }
+        server_ = RStudioGinjector.INSTANCE.getServer();
+    }
+
+    @Override
+    public String getAnchorClass() 
+    {
+        return (package_ == null) ? styles_.xtermUnsupportedHyperlink() : styles_.xtermCommand();
+    }
+
+    @Override 
+    public void onClick(){
+        if (package_ != null)
+        {
+            server_.isPackageHyperlinkSafe(package_, new ServerRequestCallback<Boolean>(){
+                
+                @Override
+                public void onResponseReceived(Boolean response) {
+                    events_.fireEvent(new SendToConsoleEvent(code_, response));
+                }
+    
+                @Override
+                public void onError(ServerError error) {}
```

Comment:
We should log errors here. (Or use `SimpleRequestCallback` to get that behavior "for free", e.g. see https://github.com/rstudio/rstudio/blob/main/src/gwt/src/org/rstudio/studio/client/common/SimpleRequestCallback.java